features/step_definitions/forms.rb

```diff
@@ -12,6 +12,12 @@
   find('label', text: text).click
 end
 
+Then(/^I click "([^"]*)" for the radio button "([^"]*)"$/) do |text, label|
+  node = find('.govuk-fieldset__heading', text: label)
```

Comment:
How about this. I think is more readable and we don't need to use class names as these can be fragile:

```ruby
Then(/^I click "([^"]*)" for the radio button "([^"]*)"$/) do |text, legend|
  find(
    'legend > span', text: legend
  ).ancestor(
    'fieldset', order: :reverse, match: :first
  ).find(
    'label', text: text
  ).click
end

Reply:
That's nice, I'm going to give it a try 👍